test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
     end
   end
 
+  describe 'create_or_update_teacherset ' do
+    it 'test create or update teacherset' do
+      ts_items_info = {bibs_resp: SIERRA_USER, total_count: 1, available_count: 1, availability_string: 'available'}
+      TeacherSet.instance_variable_set(:@req_body, SIERRA_USER["data"][0])
+      bnumber = {bnumber: "b7899158"}
+      teacher_set = TeacherSet.new(bnumber: "b7899158")
+      subjects_arr = ["Elections."]
+
+      resp = nil
+      @mintest_mock2.expect(:call, ts_items_info, [7899158])
```

Comment:
So. I realise this may be a bit annoying for you Deepika — and I honestly appreciate all the work you've put in to this — but I've come to the conclusion that this is probably too complex for a unit test. 

For a unit test we typically want to test the outputs with different inputs and _not_ worry about what happens within the method. I think where we _do_ want to use a mock there is if we're calling an external service (like an API). So `get_items_info_from_bibs_service`, `update_subjects_via_api` should be mocked, but that may be it.

That way you can just set up the ts_item_info, and then check the attributes of the TeacherSet instance created. 

Does that make sense? Sorry if this is causing a lot of extra work.